samples/1_Utils/hipDispatchLatency/hipDispatchEnqueueRateMT.cpp

```diff
@@ -126,15 +127,15 @@ struct thread_pool {
             thread.get();
         }
         threads.clear();
-        shared = {0};
+        shared.store(0, std::memory_order_seq_cst);
```

Comment:
Could you please explain why sequentially-consistent ordering was chosen here?

Reply:
It doesn't actually matter here. When this line is executed all other threads that could possibly touch this memory location are already gone. So no synchronization is needed. Any new threads that might possible start in the future will be see any store to `shared` due to implicit synchronization by creation of threads.

However, I just used seq_cst, just for safety and documentation. It is not performance critical, and it is a sample after all anyhow.

The seq_cst is actually a default for `std::atomic_int::store` method.

The shared is only really used by the worker threads to implement a thread id and a barrier, it is not actually really used for any synchronization in the `thread_pool` itself. But it must be reset, to allow new calls to `start`.